File: Data Structures & Algorithms/products-of-array-discluding-self/submission-11.java
class Solution {
    public int[] productExceptSelf(int[] nums) {

        // nums -> 1, 2, 4, 6

        // first pass -> 1, 1, 2, 8

        // second pass -> 48, 24, 12, 8

        int[] solution = new int[nums.length];
        
        int count = 1;

        for(int i = 0; i < nums.length; i++) {
            solution[i] = count;
            count *= nums[i];
        }

        count = nums[nums.length - 1];

        for(int i = nums.length - 2; i >= 0; i--) {

            solution[i] *= count;
            count *= nums[i];

        }

        return solution;
        
    }
}  

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-11.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:

        stack = []

        for token in tokens:

            # Edge case where stack is empty when computing
            if not stack and token in ["+", "-", "*", "/"]:
                return

            if token == '+':
                second = stack.pop()
                first = stack.pop()
                stack.append(first + second)

            elif token == '-':
                second = stack.pop()
                first = stack.pop()
                stack.append(first - second)

            elif token == '*':
                second = stack.pop()
                first = stack.pop()
                stack.append(first * second)

            elif token == '/':
                second = stack.pop()
                first = stack.pop()
                stack.append(int(first / second))
            else:
                stack.append(int(token))

        
        return stack[-1]